examples/structured_data/classification_with_tfdf.py

```diff
@@ -0,0 +1,669 @@
+            )
+
+        target_values = tf.cast(target_values, tf.dtypes.bool)
+
+        print("Target encoding: Computing unique feature values...")
+        # Get feature vocabulary.
+        unique_feature_values = tf.sort(tf.unique(feature_values).y)
+
+        print(
+            "Target encoding: Computing frequencies for feature values with positive targets..."
+        )
+        # Filter the data where the target label is positive.
+        positive_indices = tf.where(condition=target_values)
+        postive_feature_values = tf.gather_nd(
+            params=feature_values, indices=positive_indices
+        )
+        # Compute how many times each feature value occurred with a positive target label.
+        positive_frequency = tf.math.unsorted_segment_sum(
+            data=tf.ones(
+                shape=(postive_feature_values.shape[0], 1), dtype=tf.dtypes.int32
+            ),
+            segment_ids=postive_feature_values,
+            num_segments=unique_feature_values.shape[0],
+        )
+
+        print(
+            "Target encoding: Computing frequencies for feature values with negative targets..."
+        )
+        # Filter the data where the target label is negative.
+        negative_indices = tf.where(condition=tf.math.logical_not(target_values))
+        negative_feature_values = tf.gather_nd(
+            params=feature_values, indices=negative_indices
+        )
+        # Compute how many times each feature value occurred with a negative target label.
+        negative_frequency = tf.math.unsorted_segment_sum(
+            data=tf.ones(
+                shape=(negative_feature_values.shape[0], 1), dtype=tf.dtypes.int32
+            ),
+            segment_ids=negative_feature_values,
+            num_segments=unique_feature_values.shape[0],
+        )
+
+        print("Target encoding: Storing target encoding statistics...")
+        self.positive_frequency_lookup = tf.constant(positive_frequency)
+        self.negative_frequency_lookup = tf.constant(negative_frequency)
+
+        self._is_adapted = True
+
+    def reset_state(self):
+        if self.built:
+            self.positive_frequency_lookup = None
+            self.negative_frequency_lookup = None
+
+    def call(self, inputs):
+        # data is expected to be an integer numpy array to a Tensor shape [num_exmples, 1].
+        # This includes the feature values for a given feature in the dataset.
+
+        # Raise an error if the target encoding statistics are not computed.
+        if (
+            self.positive_frequency_lookup == None
+            or self.negative_frequency_lookup == None
+        ):
+            raise ValueError(
+                f"You need to call the adapt method to compute target encoding statistics."
+            )
+
+        inputs = self._standardize_inputs(inputs)
+
+        # Fix and validate inputs shape.
+        if inputs.shape.rank == 1:
```

Comment:
You could remove this block and just require rank 2, this is a code example and it needs to be kept short & focused